robot_driver/src/robot_driver.cpp

```diff
@@ -339,92 +298,22 @@ void RobotDriver::checkMessagesForSafety() {
 
 bool RobotDriver::updateState() {
   if (is_hardware_) {
-    // Get the newest data from the robot (BLOCKING)
-    bool fully_populated = hardware_interface_->recv(
-        last_joint_state_msg_, last_imu_msg_, user_rx_data_);
-
-    ros::Time state_timestamp = ros::Time::now();
-
-    // Check if robot data was recieved
-    if (fully_populated) {
-      last_robot_state_msg_.body.twist.angular = last_imu_msg_.angular_velocity;
-      last_robot_state_msg_.joints = last_joint_state_msg_;
-      last_joint_state_msg_.header.stamp = state_timestamp;
-      last_imu_msg_.header.stamp = state_timestamp;
+    if (estimator_id_ == "comp_filter") {
+      // Get the newest data from the robot (BLOCKING)
+      bool fully_populated = hardware_interface_->recv(
+          last_joint_state_msg_, last_imu_msg_, user_rx_data_);
+      return comp_filter_estimator_->updateState(
+          fully_populated, last_imu_msg_, last_joint_state_msg_,
+          last_mocap_msg_, last_robot_state_msg_);
+    } else if (estimator_id_ == "ekf_filter") {
+      return ekf_estimatror_->updateState();
```

Comment:
This should be state_estimator_->updateState(), where the input args are identical regardless the derived class. If they would normally have different args, perhaps we should just leave updateState() with zero args, and use other class functions (either base or derived) to load the data into members of state_estimator_ that updateState would need.

Reply:
For example, I think it would be safe to assume that updateState may take imu + joint data as an input, but if one derived class uses mocap and another doesn't, that mocap data should be provided to the derived class before calling this method.